modules/local/dvpolish/createFinalAsm.nf

```diff
@@ -2,77 +2,77 @@ process DVPOLISH_CREATE_FINALASM {
-        'nf-core/seqkit:2.8.2--h9ee0642_0' }"
+        'https://depot.galaxyproject.org/singularity/samtools:1.22.1--h96c455f_0' :
+        'biocontainers/samtools:1.22.1--h96c455f_0' }"
 
     input:
     tuple val(meta), path(unpol_fasta), path(unpol_merqury_csv)  // meta map, unpolished assembly, corresponding merqury qv file
     tuple val(meta2), path(pol_fasta), path(pol_merqury_csv)     // meta map, polished assembly, corresponding merqury qv file
 
     output:
-    tuple val(meta), path('*.fasta.gz')  , emit: fasta_gz
-    path  "versions.yml"                 , emit: versions
+    tuple val(meta), path('*.fasta.gz')     , emit: fasta_gz
+    tuple val(meta), path('*_selection.tsv'), emit: dvpolish_selection_tsv
+    path  "versions.yml"                    , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args       = task.ext.args ?: ''
     def prefix     = task.ext.prefix ?: "${meta.id}"
-    def unpol_name = unpol_fasta.getBaseName()
-    def unpol_ext  = unpol_fasta.getExtension()
-    def pol_name   = pol_fasta.getBaseName()
-    def pol_ext    = pol_fasta.getExtension()
-
     """
+    # Validate merqury files have same number of lines & are not empty
     nl_unpol_ASM=\$(wc -l < ${unpol_merqury_csv})
     nl_pol_ASM=\$(wc -l < ${pol_merqury_csv})
 
-    if [[ \${nl_unpol_ASM} -ne \${nl_pol_ASM} ]]
-    then 
-        >&2 echo "[ERROR] DVPOLISH_CREATE_FINALASM: merqury files have different lines: ${unpol_merqury_csv}: \${nl_unpol_ASM} != ${pol_merqury_csv}: \${nl_pol_ASM}"
+    if [[ \${nl_unpol_ASM} -ne \${nl_pol_ASM} ]]; then
+        >&2 echo "[ERROR] DVPOLISH_CREATE_FINALASM: merqury files have different line counts: ${unpol_merqury_csv}: \${nl_unpol_ASM} != ${pol_merqury_csv}: \${nl_pol_ASM}"
+        exit 1
+    fi
+    if [[ \${nl_unpol_ASM} -eq 0 ]]; then
+        >&2 echo "[ERROR] DVPOLISH_CREATE_FINALASM: merqury files are empty"
         exit 1
     fi
-    
-    #split unpolished assembly by sequence ID
-    seqkit split -i -O unpolished_asm ${unpol_fasta}
 
-    #split polished assembly by sequence ID
-    seqkit split -i -O polished_asm ${pol_fasta}
+    # Create selection report header
+    echo -e "contig\\tsource_asm\\tunpolished_asm_errors\\tpolished_asm_errors\\tunpolished_asm_qv\\tpolished_asm_qv" > ${prefix}_selection.tsv
+
+    # Index both assemblies
+    samtools faidx ${unpol_fasta}
+    samtools faidx ${pol_fasta}
 
-    l=1 
-    while [[ \$l -le \${nl_pol_ASM} ]]
-    do 
-        # read qv string in line p from unpolished and polished merqury files into bash array l_uasm and l_pasm respectively
-        IFS='\t' read -r -a l_uasm <<< "\$(sed -n \${l}p ${unpol_merqury_csv})"
-        IFS='\t' read -r -a l_pasm <<< "\$(sed -n \${l}p ${pol_merqury_csv})"
+    # Process contigs in single pass (merqury files = file descriptor 0 and 3)
+    line_count=0
+    while IFS=\$'\\t' read -r -a l_uasm && IFS=\$'\\t' read -r -a l_pasm <&3; do
```

Comment:
Can you use different variable names here please. The `l_uasm` and `l_pasm` is hard to understand. Perhaps also do the same with the line counts. I would prefer the names to be easier to read.